### PR TITLE
feat: check git status before bump if option.all is falsy

### LINKED ISSUE #43: [Feature request]: check if the git working tree is clean before bumping, if not, refuse to bump
### Clear and concise description of the problem

There is a scenario where you forget to commit the changed files before bumping

### Suggested solution

Check if the git working tree is clean before bumping, if not, refuse to bump

### Alternative

_No response_

### Additional context

_No response_

### Validations

- [X] Follow our [Code of Conduct](https://github.com/antfu/.github/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guide](https://github.com/antfu/contribute).
- [X] Check that there isn't already an issue that request the same feature to avoid creating a duplicate.

### PR DESCRIPTION
**Pull request by litingyes**
### Description

enable **git status check**  if `options.all` is falsy

### Linked Issues

resolved #43

### CODE REVIEW

File: src/cli/index.ts [hunk 1]
```diff
@@ -28,6 +29,10 @@ export async function main(): Promise<void> {
       process.exit(ExitCode.Success)
     }
     else {
+      if (!options.all) {
```
Comment (antfu) [inline src/cli/index.ts:32]: We should probably also add a flag like `--no-git-check` to opt-out

Comment (litingyes) [reply src/cli/index.ts:32]: `all = false` && `noGitCheck = true` ?

Comment (antfu) [reply src/cli/index.ts:32]: Yeah, like that

Comment (litingyes) [reply src/cli/index.ts:32]: okk

Comment (litingyes) [reply src/cli/index.ts]: done

**Pull request merged**